solutions/standard/variables.tf

```diff
@@ -115,8 +122,212 @@ variable "en_key_name" {
   description = "The name to give the Key which will be created for the Event Notifications. Not used if supplying an existing Key."
 }
 
+variable "cos_key_ring_name" {
+  type        = string
+  default     = "cos-key-ring"
+  description = "The name to give the Key Ring which will be created for COS. Not used if supplying an existing key or if providing `var.existing_cos_bucket_name`."
+}
+
+variable "cos_key_name" {
+  type        = string
+  default     = "cos-key"
+  description = "The name to give the Key which will be created for COS. Not used if supplying an existing key or if providing `var.existing_cos_bucket_name`."
+}
+
 variable "skip_en_kms_auth_policy" {
   type        = bool
   description = "Set to true to skip the creation of an IAM authorization policy that permits all Event Notification instances in the resource group to read the encryption key from the KMS instance."
   default     = false
 }
+
+########################################################################################################################
+# COS
+########################################################################################################################
+
+variable "create_cos_instance" {
+  description = "Set as true to create a new Cloud Object Storage instance."
+  type        = bool
+  default     = true
+}
+
+variable "existing_cos_instance_crn" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "The CRN of an existing Cloud Object Storage instance. If not supplied, a new instance will be created."
+}
+
+variable "create_cos_bucket" {
+  description = "Set as true to create a new Cloud Object Storage bucket"
+  type        = bool
+  default     = true
+}
+
+variable "existing_cos_bucket_name" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "The name of an existing bucket inside the existing Cloud Object Storage instance. If not supplied, a new bucket will be created."
+}
+
+variable "cos_destination_name" {
+  type        = string
+  description = "The name to give the IBM Cloud Object Storage destination which will be created for storage of failed delivery events."
+  default     = "COS Destination"
+}
+
+variable "cos_bucket_name" {
+  type        = string
+  description = "The bucket name in IBM cloud object storage instance. A bucket will not be created if a value is passed for `var.existing_cos_bucket_name`."
```

Comment:
The name to use when creating the Cloud Object Storage bucket for storing failed events (NOTE: bucket names are globally unique). If 'add_bucket_name_suffix' is set to true, a random 4 characters will be added to this name to help ensure bucket name is globally unique."